tests/platform/fw_utility/test_fwutil.py

```diff
@@ -0,0 +1,501 @@
+    logging.info("Check how much PSUs DUT has")
+    psu_num_out = dut.command(cmd_num_psu)
+    psu_num = 0
+    try:
+        psu_num = int(psu_num_out["stdout"])
+    except:
+        assert False, "Unable to get the number of PSUs using command '%s'" % cmd_num_psu
+
+    logging.info("Create PSU controller for testing")
+    psu_control = request.getfixturevalue("psu_ctrl")
+    if psu_control is None:
+        pytest.fail("No PSU controller for %s, skip rest of the testing in this case" % dut.hostname)
+    pdb.set_trace()
+    all_psu_status = psu_control.get_psu_status()
+    if all_psu_status:
+        for psu in all_psu_status:
+            if psu["psu_on"]:
+                psu_control.turn_off_psu(psu["psu_id"])
+                time.sleep(5)
+
+        # perform 30 seconds timeout
+        time.sleep(30)
+
+        all_psu_status = psu_control.get_psu_status()
+        if all_psu_status:
+            # turn on all psu
+            for psu in all_psu_status:
+                if not psu["psu_on"]:
+                    psu_control.turn_on_psu(psu["psu_id"])
+                    time.sleep(5)
+
+    # wait for dut to go up
+    time.sleep(20)
+
+
+def check_bios_version(version_to_install, comp_data):
+    """
+    check if bios version was updated as expected.
+    """
+    if comp_data['version'].startswith(version_to_install):
+        return SUCCESS_CODE
+    return FAILURE_CODE
+
+
+def check_cpld_version(version_to_install, comp_data):
+
+    # for now just checks if ends without errors - without version check
+    return SUCCESS_CODE
+
+    # counts = {}
+    # ver = comp_data['version']
+    # for v in ver:
+    #     counts[v] = ver.count(v)
+
+    # if counts == version_to_install:
+    #     return SUCCESS_CODE
+
+    # return FAILURE_CODE
+
+
+def execute_install(request, dut, component, path_to_install, version_to_install, components_list):
+    cmd = "fwutil install chassis component {} fw -y {}".format(component, path_to_install)
+
+    loganalyzer = LogAnalyzer(ansible_host=dut, marker_prefix='acl')
+    loganalyzer.load_common_config()
+    try:
+        loganalyzer.except_regex = [FW_INSTALL_SUCCESS_LOG]
+        with loganalyzer:
+            stdout = dut.command(cmd)
+    except LogAnalyzerError as err:
+        raise err
+
+    if stdout['rc'] != SUCCESS_CODE:
+        raise stdout['stderr']
+
+    # complete fw update - cold reboot if BIOS, power cycle with 30 sec timeout if CPLD
+    globals()[component.lower() + '_update'](request, dut)
+    # check output of show command
+    pdb.set_trace()
+    fw_data = get_output_data(dut, components_list)
+    comp_data = fw_data[component]
+    if not comp_data['version']:
+        pytest.fail("Installation didn't work. Aborting!")
+
+    return globals()['check_' + component.lower() + '_version'](version_to_install, comp_data)
+
+
+@pytest.fixture(scope='function')
+def install_local(request, testbed_devices, testbed, get_fw_path, components_list):
```

Comment:
@noaOrMlnx can we slightly improve the entire infrastructure and move all that fixtures to the conftest file just to avoid test environment pollution? What do you think?